### PR TITLE
[Core] Adding kernel library list

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**
In order to skip tests depending on library disponibility

**🆕 Changelog**
- Adding kernel library list
- Expose to python
- Added test
- Reactivate CadModeler if triangle is available

### CODE REVIEW

File: kratos/sources/kernel.cpp [hunk 1]
```diff
@@ -50,19 +55,56 @@ void Kernel::Initialize() {
...
     else
         return false;
 }
 
+bool Kernel::IsLibraryAvailable(const std::string& rLibraryName) const {
+    if (GetLibrayList().find(rLibraryName) != GetLibrayList().end()) {
+        return true;
+    } else {
+        return false;
+    }
```
Comment (philbucher) [inline kratos/sources/kernel.cpp]:
```suggestion
    return GetLibrayList().find(rLibraryName) != GetLibrayList().end();
```

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee we believe that the idea is very good and useful. Nevertheless, we see that having them all in the kernel is not the best way to maintain them.

We think that a better mechanism would be to use the new Registry under "third_party_libraries" and add each of them in their relative wrappers. (for example,  json in our Parameters class)

**Comment by loumalouomega** [comment]
> @KratosMultiphysics/technical-committee we believe that the idea is very good and useful. Nevertheless, we see that having them all in the kernel is not the best way to maintain them.

I did it in order to be consistent with the applications

> We think that a better mechanism would be to use the new Registry under "third_party_libraries" and add each of them in their relative wrappers. (for example, json in our Parameters class)

1- Is the Registry already merged?
2- I think not all libraries have a wrapper, F.E triangle is called in many places. The current structure using Registry should work fine

**Comment by loumalouomega** [comment]
What is the status of this¿? https://github.com/orgs/KratosMultiphysics/teams/technical-committee

**Comment by loumalouomega** [comment]
> What is the status of this¿? [github.com/orgs/KratosMultiphysics/teams/technical-committee](https://github.com/orgs/KratosMultiphysics/teams/technical-committee)

👋

**Comment by loumalouomega** [comment]
I still think this is interesting

**Comment by loumalouomega** [comment]
> I still think this is interesting

@KratosMultiphysics/technical-committee

**Comment by loumalouomega** [comment]
> > I still think this is interesting
> 
> @KratosMultiphysics/technical-committee

👋

**Comment by roigcarlo** [comment]
Sorry we forgot to answer. 
1 - Yes the registry is in place and usable.
2 - Answering for myself ( not TC ): For the ones that are used everywhere I suppose there is no other option than leave them there...

**Comment by loumalouomega** [comment]
> Sorry we forgot to answer. 1 - Yes the registry is in place and usable. 2 - Answering for myself ( not TC ): For the ones that are used everywhere I suppose there is no other option than leave them there...

But here no registry is used...

**Comment by roigcarlo** [comment]
Yep, that why we asked you to use the registry :smile:

**Comment by loumalouomega** [comment]
> Yep, that why we asked you to use the registry 😄

But it is a list, I am not using classes. What's the point?

**Comment by roigcarlo** [comment]
That it becomes available everywhere without depending on having a reference to the kernel. Which btw makes sense because that info is not related to the kernel.

**Comment by loumalouomega** [comment]
> That it becomes available everywhere without depending on having a reference to the kernel. Which btw makes sense because that info is not related to the kernel.

Okay, in that case I would need to study this

**Comment by loumalouomega** [comment]
> > That it becomes available everywhere without depending on having a reference to the kernel. Which btw makes sense because that info is not related to the kernel.
> 
> Okay, in that case I would need to study this

Ups, it looks a bit of work... and create a new class

**Comment by RiccardoRossi** [comment]
this one is delegated to @roigcarlo, who is requesting to use the registry as a tool for querying for available libraries

**Comment by loumalouomega** [comment]
> this one is delegated to @roigcarlo, who is requesting to use the registry as a tool for querying for available libraries

Okay

**Comment by loumalouomega** [comment]
(just merging master, this branch wasn't merged from long time ago, I will take a look at @roigcarlo suggestion later)

**Comment by loumalouomega** [comment]
SHould be fine now @roigcarlo

**Review by roigcarlo — APPROVED** [review]

**Comment by loumalouomega** [comment]
BTW, there is any reason why this is empty? https://github.com/KratosMultiphysics/Kratos/blob/06eeea305b06c8df084b031ec6415c66e4ed43d4/kratos/sources/kernel.cpp#L63

**Pull request merged**